dropwizard-logging/src/main/java/io/dropwizard/logging/AbstractAppenderFactory.java

```diff
@@ -111,6 +115,10 @@
 
     private int discardingThreshold = -1;
 
+    @Nullable
+    @DecimalMin(value="0", inclusive = false)
+    private Double maxMessagesPerSecond;
```

Comment:
does it make sense to support a fractional number of messages per second instead of making this an `Integer` or even an `OptionalInt`?

Reply:
Yeah if you want 30 messages per minute, you'd set to this `0.5`

Reply:
It’s too bad a Duration wouldn’t work here somehow

Reply:
I think you could be on to something, you could have it like (may not be the best name):

```yaml
messageThrottle: 1ms
```

Here you'd be throttling logging to 1 message per millisecond.

Reply:
Exactly what I thought. Duration has a count and time unit which we need in this case even though it’s not the best name.